base_layer/core/src/base_node/service/error.rs

```diff
@@ -21,15 +21,17 @@
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 use crate::base_node::comms_interface::CommsInterfaceError;
-use derive_error::Error;
 use tari_comms_dht::outbound::DhtOutboundError;
+use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum BaseNodeServiceError {
-    CommsInterfaceError(CommsInterfaceError),
-    DhtOutboundError(DhtOutboundError),
-    #[error(msg_embedded, no_from, non_std)]
+    #[error("Comms interface error")]
```

Comment:
```suggestion
    #[error("Comms interface error: {0}")]
```

same for most of the rest